src/DocumentationBrowserViewExtension/DocumentationBrowserViewExtension.cs

```diff
@@ -208,13 +208,19 @@ private void OnInsertFile(object sender, InsertDocumentationLinkEventArgs e)
 
             if (!DynamoSelection.Instance.Selection.Any()) return;
 
-            GroupInsertedGraph(existingGroups, e.Name);
-            DoEvents();
-
-            // We have selected all the nodes and notes from the inserted graph
-            // Now is the time to auto layout the inserted nodes
-            this.DynamoViewModel.GraphAutoLayoutCommand.Execute(null);
-            this.DynamoViewModel.FitViewCommand.Execute(false);
+            Dispatcher.CurrentDispatcher.BeginInvoke(() =>
+            {
+                GroupInsertedGraph(existingGroups, e.Name);
```

Comment:
we do not actually change UI elements directly through `GroupInsertedGraph` ?
are you using the dispatcher just to insure proper order of execution with the 2 commands invoked bellow ?
```
this.DynamoViewModel.GraphAutoLayoutCommand.Execute(null);
this.DynamoViewModel.FitViewCommand.Execute(false);
```

Reply:
This code probably triggers all the appropriate layout updates/renders, so no need for the artificial DoEvents right ?

Reply:
yes, I am using this code to make sure the render is done before the layout.